### PR TITLE
Added Lifecycle Template

### PR DESCRIPTION
**Pull request by kdivya153**

### CODE REVIEW

File: src/sdks/core/src/cpp/templates/Lifecycle/src/module_impl.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+static void readyDispatcher(const void* result) {
+    Metrics::MetricsImpl::ready();
+}
+/* ready - Notify the platform that the app is ready */
+void ${info.Title}Impl::ready( Firebolt::Error *err ) 
+{
+    std::cout << "******Divya Index.cpp ready function for Lifecycle******" << std::endl;
+    /* TODO add prioritize() to prioritize the events right now manually updating the state*/
+
+    setState("foreground");
```
Comment (kevinshahfws) [inline src/sdks/core/src/cpp/templates/Lifecycle/src/module_impl.cpp:22]: Why is this set as foreground?

File: src/sdks/core/src/cpp/templates/Lifecycle/src/module_impl.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+/* state - return the state of the app */
+std::string LifecycleImpl::state() {
+    std::cout << "******Divya Index.cpp state function for Lifecycle******" << std::endl;
+    std::cout << "******CURRENT STATE******" << currentState << std::endl;
+    return currentState;
+}
+
+/* finished - Notify the platform that the app is done unloading */
+void ${info.Title}Impl::finished( Firebolt::Error *err )
```
Comment (kevinshahfws) [inline src/sdks/core/src/cpp/templates/Lifecycle/src/module_impl.cpp:54]: I don't see **_unloading_** condition here

File: src/sdks/core/src/cpp/templates/Lifecycle/src/module_impl.cpp [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    return;
+}
+
+/* state - return the state of the app */
+std::string LifecycleImpl::state() {
+    std::cout << "******Divya Index.cpp state function for Lifecycle******" << std::endl;
+    std::cout << "******CURRENT STATE******" << currentState << std::endl;
+    return currentState;
```
Comment (kevinshahfws) [inline src/sdks/core/src/cpp/templates/Lifecycle/src/module_impl.cpp:50]: How various states will be stored here?

File: src/sdks/core/src/cpp/sdk/cpptest/CoreSDKTest.cpp [hunk 4]
```diff
@@ -715,7 +716,41 @@ EnumMap<Firebolt::Lifecycle::LifecycleEventSource> lifecycleEventSourceMap = {
     { Firebolt::Lifecycle::LifecycleEventSource::REMOTE, "remote" }
 };
 
-void CoreSDKTest::OnBackgroundNotification::onBackground(const Firebolt::Lifecycle::LifecycleEvent& lifecycleEvent)
+void CoreSDKTest::LifecycleReady()
```
Comment (kschrief) [inline src/sdks/core/src/cpp/sdk/cpptest/CoreSDKTest.cpp:719]: Do these tests fail/throw an error when we don't get back a proper response?

File: src/sdks/core/src/cpp/templates/Lifecycle/src/module_impl.cpp [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+
+/* ready - Notify the platform that the app is ready */
+void ${info.Title}Impl::ready(Firebolt::Error *err) {  
+    Firebolt::Error status = Firebolt::Error::NotConnected;
+
+    JsonObject jsonParameters;
+
+    // Call Prioritize with  callback
+    status = FireboltSDK::Event::Instance().Prioritize<JsonData_LifecycleEvent>("lifecycle.onForeground", jsonParameters, onReadyInnerCallback, (void*)nullptr, this);
```
Comment (kschrief) [inline src/sdks/core/src/cpp/templates/Lifecycle/src/module_impl.cpp:74]: What is this doing?

**Comment by kdivya153** [comment]
Duplicate of : https://github.com/rdkcentral/firebolt-apis/pull/292

**Pull request closed without merging**